src/App/Repository/HXL/HXLTagRepository.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * Ushahidi HXLTag Repository, using Kohana::$config
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Repository\HXL;
+
+use Ohanzee\Database;
+use Ushahidi\Core\SearchData;
+use Ushahidi\Core\Entity\HXL\HXLTag;
+use Ushahidi\Core\Entity\HXL\HXLTagRepository as HXLTagRepositoryContract;
+use Ushahidi\Core\Usecase\ReadRepository;
+use Ushahidi\Core\Usecase\SearchRepository;
+use Ushahidi\App\Repository\OhanzeeRepository;
+use Ohanzee\DB;
+
+class HXLTagRepository extends OhanzeeRepository implements
+	HXLTagRepositoryContract,
+	SearchRepository,
+	ReadRepository
+{
+	public function __construct(
+		Database $db
+	) {
+		parent::__construct($db);
+	}
+
+	// OhanzeeRepository
+	protected function getTable()
+	{
+		return 'hxl_tags';
+	}
+
+	public function getSearchFields()
+    {
+		return ['tag_name'];
+	}
+
+	public function setSearchConditions(SearchData $search)
+    {
+        $query = $this->search_query;
+        return $query;
+    }
+
+    public function getEntity(array $data = null)
+    {
+    	if ($data) {
+			$data['form_attribute_types'] = array_flatten($this->getFormAttributeTypes($data['id']));
+			$data['hxl_attributes'] = $this->getHXLAttributes($data['id']);
+    	}
+		return new HXLTag($data);
+    }
+
+	/**
+	 * @param $hxl_tag_id
+	 */
+	protected function getHXLAttributes($hxl_tag_id)
+	{
+		return DB::select('hxl_attributes.id', 'hxl_attributes.attribute', 'hxl_attributes.description')
```

Comment:
Is the likely usecase for this just loading all HXL tags at once and then their attributes?

If so, could we query *all* hxl_tag_attributes, joined to their attributes. Then cache that (ie. into $this->tag_attributes) and just return values from there as needed? ie. reduce from 1 query per tag, to 1 query.

Reply:
That's be   better yea